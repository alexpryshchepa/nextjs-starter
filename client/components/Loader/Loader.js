import React from 'react';
import PropTypes from 'prop-types';
import cssVars from 'client/styles/variables.scss';
import s from './Loader.scss';

export const Loader = ({ size, color }) => (
  <div className={s.root}>
    <span
      className={s.spinner}
      style={{
        width: `${size}px`,
        height: `${size}px`,
        borderColor: color,
        borderRightColor: 'transparent',
      }}
    />
  </div>
);

Loader.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
};

Loader.defaultProps = {
  size: 24,
  // Usage of sharing css variables between css and js
  color: cssVars['color-primary'],
};

export default Loader;
