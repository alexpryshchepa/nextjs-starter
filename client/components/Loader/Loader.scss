.root {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  height: 100%;
  cursor: default;
  opacity: 0.65;
  overflow: hidden;

  @keyframes spin {
    0% {
      transform: rotate(0deg);
    }

    100% {
      transform: rotate(360deg);
    }
  }
}

.spinner {
  width: 24px;
  height: 24px;
  border-width: 3px;
  border-style: solid;
  border-radius: 50%;
  animation: spin 0.7s linear infinite;
}
