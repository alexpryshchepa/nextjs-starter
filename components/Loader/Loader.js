import React from 'react';
import PropTypes from 'prop-types';
import s from './Loader.scss';

const Loader = ({ size, color }) => (
  <div className={s.root}>
    <span
      className={s.spinner}
      style={{
        width: `${size}px`,
        height: `${size}px`,
        borderColor: color,
        borderRightColor: 'transparent',
      }}
    />
  </div>
);

Loader.propTypes = {
  size: PropTypes.number,
  color: PropTypes.string,
};

Loader.defaultProps = {
  size: 24,
  color: '#000',
};

export default Loader;
