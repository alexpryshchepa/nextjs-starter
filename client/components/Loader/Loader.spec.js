import { shallow } from 'enzyme';
import React from 'react';
import { Loader } from './Loader';

describe('loader component', () => {
  const component = shallow(
    <Loader
      size={32}
      color="#f00"
    />,
  );

  it('renders loader', () => {
    expect(
      component.find('span.spinner').prop('style'),
    ).toEqual({
      width: '32px',
      height: '32px',
      borderColor: '#f00',
      borderRightColor: 'transparent',
    });
  });

  it('checks snapshot matching', () => {
    expect(component).toMatchSnapshot();
  });
});
