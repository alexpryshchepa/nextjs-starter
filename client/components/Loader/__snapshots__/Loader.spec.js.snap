// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loader component checks snapshot matching 1`] = `ShallowWrapper {}`;
